Sources/CvPlayer.cpp

```diff
@@ -6065,21 +6023,16 @@ bool CvPlayer::hasBusyUnit() const
 {
 	PROFILE_FUNC();
 
-	CvSelectionGroup* pLoopSelectionGroup;
-	int iLoop;
-
-	for(pLoopSelectionGroup = firstSelectionGroup(&iLoop); pLoopSelectionGroup; pLoopSelectionGroup = nextSelectionGroup(&iLoop))
+	for (CvPlayer::group_iterator groupItr = beginGroups(); groupItr != endGroups(); ++groupItr)
 	{
-		if (pLoopSelectionGroup->isBusy())
+		CvSelectionGroup* pLoopGroup = *groupItr;
+		if (pLoopGroup->isBusy() && pLoopGroup->getNumUnits() == 0)
 		{
-		    if (pLoopSelectionGroup->getNumUnits() == 0)
-		    {
-		        pLoopSelectionGroup->kill();
-		        return false;
-		    }
-
-			return true;
+			pLoopGroup->kill();
+			return false;
 		}
+
+		return true;
 	}
```

Comment:
Do I need to use the other iterator here?

Reply:
No that one looks correct?